sql/core/src/test/scala/org/apache/spark/sql/execution/RowFormatConvertersSuite.scala

```diff
@@ -87,4 +91,36 @@ class RowFormatConvertersSuite extends SparkPlanTest {
       input.map(Row.fromTuple)
     )
   }
+
+  test("SPARK-9683: we should deep copy UTF8String when convert unsafe row to safe row") {
```

Comment:
This PR is to fix the problem for UnsafeArrayData, not UnsafeRow, we should also update the JIRA and title.